### PR TITLE
The fourth part of  td-shim refactor

### PR DESCRIPTION
**Pull request by jiangliu**
1) combine td-shim-ld/td-shim-enroll-key/td-shim-sign-payload into a unified crate td-shim-tools
2) define all td-shim spec defined constants/structure in td-shim.

### CODE REVIEW

File: doc/tdshim_spec.md [hunk 1]
```diff
@@ -267,6 +267,14 @@ typedef enum {
...
   PayloadImageTypeRawVmLinux,
 
+  // Payload Binary is vmlinux, follow Linux boot protocol.
+  // It's an ELF64 binary image.
+  PayloadImageTypeVmLinux,
+
+  // Auto-detect type of the payload image, follow Linux boot protocol.
+  // It's one of bzImage, vmLinux. RawVmLinux should be explicitly marked
+  // instead of auto-detection.
+  PayloadImageTypeAutoDetect,
```
Comment (jyao1) [inline doc/tdshim_spec.md:277]: Should it be align with latest spec in main?

Comment (jiangliu) [reply doc/tdshim_spec.md:277]: Let first merge the spec, then update the code:)

Comment (jyao1) [reply doc/tdshim_spec.md:277]: i mean the spec should be chanaged - PayloadImageTypeAutoDetect should be removed

Comment (jiangliu) [reply doc/tdshim_spec.md:277]: Sorry, the last patch should be removed:) 
Fixed.

**Pull request merged**